tests/unit/maxmemory.tcl

```diff
@@ -3,100 +3,130 @@ start_server {tags {"maxmemory"}} {
     r config set maxmemory-policy allkeys-lru
     set rd [redis_deferring_client]
 
-    proc refill {} {
+    proc init_test {client_eviction} {
         r flushdb
+
+        set prev_maxmemory_clients [r config get maxmemory-clients]
+        if $client_eviction {
+            r config set maxmemory-clients 3mb
+        } else {
+            r config set maxmemory-clients 0
+        }
+        # If maxmemory-clients changed we need to wait for clients cron to update all the clients accordingly
+        if {$prev_maxmemory_clients != [r config get maxmemory-clients]} {
+            after 1500
+        }
+
+        r config resetstat
         # fill 5mb using 50 keys of 100kb
         for {set j 0} {$j < 50} {incr j} {
             r setrange $j 100000 x
         }
+        assert {[r dbsize] == 50}
+    }
+    
+    proc verify_test {client_eviction} {
+        set evicted_keys [s evicted_keys]
+        set evicted_clients [s evicted_clients]
+        set dbsize [r dbsize]
+        
+        puts "evicted clients: $evicted_clients, evicted keys: $evicted_keys, dbsize: $dbsize"
+        if $client_eviction {
+            assert {$evicted_clients > 0 && $evicted_keys == 0 && $dbsize == 50}
+        } else {
+            assert {$evicted_clients == 0 && $evicted_keys > 0 && $dbsize < 50}
```

Comment:
i'd rather use assert_equal and assert_range in separate lines.
the difference is that when they fail, we see the value (the assert proc gets it as an argument, and not just a boolean).
also, it could be nice to add an `if {$::verbose}` that will print these, so that we know how close we usually get to the threshold, and see if we need to tune it.

Reply:
done.